### ISSUE TITLE
Clear customized item will throw error if create a new one when item list is empty 

### ISSUE THREAD

**Issue by sheguloves** [angular-ui/ui-select#2058]
In my usage, I create a customized tag then clear it. after that, i clicked input component, it throw a error at following code:
```
function _resetSearchInput() {
    if (ctrl.resetSearchInput) {
      ctrl.search = EMPTY_SEARCH;
      //reset activeIndex
      if (ctrl.selected && ctrl.items.length && !ctrl.multiple) {
        ctrl.activeIndex = _findIndex(ctrl.items, function(item){
          return angular.equals(this, item);
        }, ctrl.selected);
      }
   }
}
```
"ctrl.items" is undefined.
when clear selection in directive "uiSelectMultiple", it calls "ctrl.refreshComponent", as following show: 
```
ctrl.refreshComponent = function(){
    //Remove already selected items
    //e.g. When user clicks on a selection, the selected array changes and
    //the dropdown should remove that item
    if($select.refreshItems){
      $select.refreshItems();
    }
    if($select.sizeSearchInput){
      $select.sizeSearchInput();
    }
};
```
"$select.refreshItems" pass undefined to function, finally go into "setPlainItems(items)", but "items" is undefined. Maybe "setPlainItems(items)" should change as following:
```
function setPlainItems(items) {
  ctrl.items = items || [];
}
```

**Comment by dustin-page** [comment]
I have the same issue. Just ran into it today

**Comment by dustin-page** [comment]
@sheguloves 
I figured out the problem I was having; maybe you are having the same issue. In my case, I have 2 custom validation directives that are validating the ui-select ng-model data; 'ui-select-validate-email' and 'ui-select-required'.

It turns out that AngularJS sets the $modelValue to undefined when a value doesn't validate correctly. This breaks the code in ui-select because the model becomes undefined instead of reverting back to an empty array.

 I fixed the issue by adding model-options="{ allowInvalid: true }" on the ui-select directive:

Example:
`<ui-select 
           multiple
           tagging="true"
           tagging-label="(New Email Address)"
           ng-model="vm.emails"
           ng-model-options="{ allowInvalid: true }"
           ui-select-validate-email="true"
           ui-select-required="true"
           theme="bootstrap">`

**Comment by sheguloves** [comment]
@dustin-page 
Yes, invalid value will change ng-model undefined. Thanks for your reply
